clusters.cnp.example.org/definition.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: CompositeResourceDefinition
+metadata:
+  name: clusters.cnp.example.org
+spec:
+  defaultCompositionRef:
+    name: clusters.cnp.example.org
+  # This policy is equivalent to the existing InfrastructurePublication. Making
+  # this a field of the InfrastructureDefinition implies that the package author
+  # controls the publication of packaged types; it cannot easily be overridden
+  # without forking/duplicating the package. The options are 'Private' (the
+  # default), or 'Published'.
+  publicationPolicy: Published
```

Comment:
I suggest renaming this. Jan and Tim were both confused by the work "publication" showing up here, they thought this either applied to the package or thought this interacted with K8s RBAC in some way. 

Perhaps we could rename this to "scope: cluster/namespace/private" or "enable: true/false" would be more clear?

Reply:
I've always been a fan of `scope: [cluster | namespace]` or `scope: [public | private |internal]`.

However here `publicationPolicy` is controlling if a namespace-scoped CRD is published **in addition to** the cluster-scoped CRD that is added to the API when the `CompositeResourceDefinition` is created. 

Given that, suggest replacing `publicationPolicy` with something more obvious like `publishNamespaceCRD: [true | false]` which if omitted would default to false.

Reply:
Let's not make this a boolean. It's [an anti-pattern that prevents us ever extending the field to support additional options](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types)

I also don't think we can frame this as "is it cluster scoped or namespaced" given that it's additive. It's actually "is it only cluster scoped, or is there also a namespaced proxy CR?"

Reply:
Followup issue:  https://github.com/crossplane/example-cnp/issues/15